server/chat-plugins/datasearch.js

```diff
@@ -747,6 +773,33 @@ function runDexsearch(target, cmd, canAll, message) {
 	let resultsStr = (message === "" ? message : `<span style="color:#999999;">${escapeHTML(message)}:</span><br />`);
 	if (results.length > 1) {
 		results.sort();
+		if (order) {
+			let stat = order.substr(0, order.length - 1);
+			let sort = order[order.length - 1];
+			results.sort((a, b) => {
+				let mon1 = mod.getTemplate(sort === '+' ? a : b), mon2 = mod.getTemplate(sort === '+' ? b : a);
+				let monStat1, monStat2;
```

Comment:
This needs to be initialized to `0` if you want the BST calculation to work correctly.